test/DynamoCoreWpfTests/SearchSideEffects.cs

```diff
@@ -87,5 +80,63 @@ public void WhenCustomNodeWorkspaceIsFocusedInputAndOutputNodesArePresentInSearc
 
         }
+
+        /// <summary>
+        /// This test will validate that the nodes "Input", "Output", "And", "Or", "Not", "+", "-"  appear in the InCanvasSearch results
+        /// </summary>
+        [Test]
+        [Category("UnitTests")]
+        public void WhenStartingDynamoOperatorNodesNolongerMissingFromSearch()
+        {
+            Assert.IsAssignableFrom(typeof(HomeWorkspaceModel), ViewModel.Model.CurrentWorkspace);
+
+            List<string> nodesList = new List<string>() { "Input", "Output", "And", "Or", "Not", "+", "-" };
+
+            foreach (var node in nodesList)
+            {
+                // search and check that the results are correct based in the node name provided for the searchTerm
+                ViewModel.CurrentSpaceViewModel.InCanvasSearchViewModel.SearchAndUpdateResults(node);
+                var filteredResults = ViewModel.CurrentSpaceViewModel.InCanvasSearchViewModel.FilteredResults;
+                Assert.AreEqual(1, filteredResults.Count(x => x.Model.Name == node), "Non matching results for " + node);
+            }
+        }
+
+        //This test will validate that all the nodes read by Dynamo in test mode can be found using LuceneSearch
+        [Test]
+        [Category("UnitTests")]
+        public void LuceneSearchAllNodesValidation()
+        {
+            Assert.IsAssignableFrom(typeof(HomeWorkspaceModel), ViewModel.Model.CurrentWorkspace);
+            List<string> nodesList = ViewModel.Model.SearchModel.SearchEntries.Select(entry => entry.Name).ToList();
+
+            foreach (var node in nodesList)
+            {
+                // Search and check that the results are correct based in the node name provided for the searchTerm
+                ViewModel.CurrentSpaceViewModel.InCanvasSearchViewModel.SearchAndUpdateResults(node);
+                var filteredResults = ViewModel.CurrentSpaceViewModel.InCanvasSearchViewModel.FilteredResults;
+
+                //There are overloaded nodes
+                Assert.GreaterOrEqual(filteredResults.Count(x => x.Model.Name == node), 1);
+            }
+        }
+
+        //This test will validate that when the search term is a category it will found the nodes under that specific category
+        [Test]
+        [Category("UnitTests")]
+        public void LuceneSearchNodesByCategoryValidation()
+        {
+            Assert.IsAssignableFrom(typeof(HomeWorkspaceModel), ViewModel.Model.CurrentWorkspace);
+            string category = "Core.Input";
+
+            // Search and check that the results are correct based in the node name provided for the searchTerm
+            var nodesResult = ViewModel.CurrentSpaceViewModel.InCanvasSearchViewModel.Search(category,true);
+
+            Assert.That(nodesResult.Count(), Is.GreaterThan(0));
+
+            foreach (var node in nodesResult.Take(10))
+            {
+                Assert.IsTrue(node.Category.Contains(category.Split('.')[0]) || node.Category.Contains(category.Split('.')[1]));
```

Comment:
Comments for this line please

Reply:
I've added more comments in the next commit: https://github.com/DynamoDS/Dynamo/pull/14401/commits/a4f79e14156fbcde266db4eddee69b591bd345a8
Please review it. Thanks